### PR TITLE
Dev #369 closures to wollok

### PR DESCRIPTION
**Pull request by javierfernandes**
Closures are now developed in wollok itself.

### CODE REVIEW

**Comment by npasserini** [comment]
Zarpado!

File: org.uqbar.project.wollok.lib/src/wollok.wlk [hunk 1]
```diff
@@ -512,6 +520,16 @@ package lang {
...
 		override method internalToSmartString(alreadyShown) = start.toString() + ".." + end.toString()
 	}
+	
+	/**
+	 * @author jfernandes
+	 * @since 1.3
+	 * @noInstantiate
+	 */
+	class Closure {
+		method apply(parameters...) native
```
Comment (javierfernandes) [inline org.uqbar.project.wollok.lib/src/wollok.wlk]: eventually we will have more methods here like "curry" and maybe even some almost reflectives like "getParameterNames()" ?

**Comment by javierfernandes** [comment]
That was the last native object migrated to wollok.
We have now the path clear to perform a big semantic refactoring in the interpreter.
All evaluations should change from this:

```xtend
    def Object evaluate(EObject e)
```

To

```xtend
    def WollokObject evaluate(EObject e)
```

Also we can delete all the code for handling java ints, strings, etc.. in the runtime.
There should be only WollokObject.
All arguments to methods calls can also be considered as WollokObjects.

That should reduce the code.

**Comment by npasserini** [comment]
Great!

**Pull request merged**